blocks.js

```diff
@@ -3484,6 +3484,11 @@ BlockMorph.prototype.destroy = function () {
     this.allComments().forEach(function (comment) {
         comment.destroy();
     });
+    
+    if (this.topBlock() === this && this.activeProcess()) {
```

Comment:
Just a really tiny quibbly, maybe totally irrelevant:
Looks like we're scanning for topBlock several times, especially when using the context menu, which can be a potential strain if the block is somewhere way at the bottom of a giant script (I've seen those ^^). Instead of scanning for the topBlock and comparing it to ```this``` one can't we just assume that if the current block either doesn't have a Morphic parent or that parent doesn't know the ```topBlock``` method (e.g. if it's a Hand or a ScriptsMorph), then it must be a topBlock itself?

```JavaScript
if (!this.parent || !this.parent.topBlock) {
        // I must be a top block myself
        this.activeProcess().stop();
}
```

Okay, up to you, fine with me :-)